### PR TITLE
Add dialog for all VDO converted.

### PR DESCRIPTION
**Pull request by jshimkus-rh**
Skips checking devices.

### CODE REVIEW

**Comment by pirat89** [comment]
To be honest, I am more thinking about merging both dialogs into the one, raised in the checkphase. Let's discuss it later. running home \o

**Comment by jshimkus-rh** [comment]
I get that.  There's quite a bit of effective overlap between the two.

**Review by smitterl — CHANGES_REQUESTED** [review]
Hey, this works for me. There's a minor typo I think we should definitely fix (s. comments).

I'm also wondering if we can't add any information for users to identify for themselves if they even have vdo type volumes. From [RHEL 8 docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/deduplicating_and_compressing_logical_volumes_on_rhel/creating-a-deduplicated-and-compressed-logical-volume_deduplicating-and-compressing-logical-volumes-on-rhel#creating-an-lvm-vdo-volume_creating-a-deduplicated-and-compressed-logical-volume) I hoped to find information via manpage lvmvdo but I can't seem to find the manpage and am not really sure how I could check this. Does somebody know more?

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py [hunk 1]
```diff
@@ -32,19 +32,60 @@ class CheckVdo(Actor):
...
+                    label='Are all VDO devices succesfully converted to LVM '
+                          'management?',
+                    description='Enter True if all VDO devices on the system '
+                                'have been successfully converted to LVM '
+                                'management.  Entering True will circumvent '
+                                'checking block devices states as VDO devices '
+                                'in need of conversion.  All VDO devices must '
+                                'be converted to LVM management before '
+                                'upgrading.',
+                    reason='To maximze safety all block devices on a system '
```
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: Fixed.

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py [hunk 2]
```diff
@@ -32,19 +32,60 @@ class CheckVdo(Actor):
...
+                           'that meet the criteria as possible VDO devices are '
+                           'checked to verify that, if VDOs, they have been '
+                           'converted to LVM management.  If the check of any '
```
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: Not clear on the issue unless it's the double spaces after '.'?
But I did notice some over 80 situations which I addressed.

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py [hunk 3]
```diff
@@ -32,19 +32,60 @@ class CheckVdo(Actor):
...
+                           'device fails for any reason an upgrade  inhibiting '
```
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: Same response as to previous "Too many blanks?"

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: As to identifying vdo devices...
(note that sdb is a pre-conversion vdo and sdc is post-conversion)
lsblk proudces the following...
`# lsblk

NAME                          MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
sdb                             8:16   0   16G  0 disk 
└─vdo-sdb                     253:3    0   12G  0 vdo  
sdc                             8:32   0   16G  0 disk 
└─vdovg-vdolvol_vpool_vdata   253:4    0   16G  0 lvm  
  └─vdovg-vdolvol_vpool-vpool 253:5    0   12G  0 lvm  
    └─vdovg-vdolvol           253:6    0   12G  0 lvm  `

And 'dmsetup status' produces...
`# dmsetup status

vdo-sdb: 0 25116416 vdo /dev/sdb normal - online online 1050700 4194304
vdovg-vdolvol: 0 25116416 linear 
vdovg-vdolvol_vpool-vpool: 0 25116416 vdo /dev/dm-4 normal - online online 1050188 4193792
vdovg-vdolvol_vpool_vdata: 0 33550336 linear `

But both of those are with the vdo device active.

[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: blkid produces identifying info regardless of the vdo device being active...

`# blkid

/dev/sdb: UUID="9a5cf4a3-8264-48a3-a529-f0b2908d5615" TYPE="vdo"
/dev/sdc: UUID="eJ40K2-d5uF-Ux8g-JBBl-Hh94-u4Ce-XM26vq" TYPE="LVM2_member"`

I was going to include a warning that such identification only worked if the vdo packages were installed and if they'd been uninstalled that blkid no longer identified 'TYPE=vdo' devices.  But in verifying that it's apparently not (or no longer; I'd swear I'd seen it in the past) the case.

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (smitterl) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: `upgrade  inhibiting`

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (smitterl) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: > # lsblk
> 
> NAME MAJ:MIN RM SIZE RO TYPE MOUNTPOINT sdb 8:16 0 16G 0 disk └─vdo-sdb 253:3 0 12G 0 vdo sdc 8:32 0 16G 0 disk └─vdovg-vdolvol_vpool_vdata 253:4 0 16G 0 lvm └─vdovg-vdolvol_vpool-vpool 253:5 0 12G 0 lvm

Thank you. What about we add a note "E.g. you can see if you are using vdo at all by running `lsblk|grep vdo`; an empty output will confirm there are no vdo volumes."

**Review by smitterl — APPROVED** [review]

**Comment by smitterl** [comment]
@jshimkus-rh The PR is in "Draft" status, is this on purpose?

**Comment by pirat89** [comment]
> @jshimkus-rh The PR is in "Draft" status, is this on purpose?

yes, we are discussing the right solution and most probably we will go with a different one where we merge both dialogs.

**Comment by jshimkus-rh** [comment]
> On Aug 29, 2022, at 07:02, Petr Stodůlka ***@***.***> wrote:
> 
> ﻿
> @jshimkus-rh The PR is in "Draft" status, is this on purpose?
> 
> yes, we are discussing the right solution and most probably we will go with a different one where we merge both dialogs.
> 

Petr, did you see my email re the single dialog?  Perhaps I missed them?
I wanted your comments before making a PR.

File: repos/system_upgrade/el8toel9/actors/checkvdo/actor.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jshimkus-rh) [reply repos/system_upgrade/el8toel9/actors/checkvdo/actor.py]: Saw that after, but the reflow for >80 cols cleaned it up.

**Comment by jshimkus-rh** [comment]
Closing this PR.  New PR  #961 collapsing to a single dialog is  its replacement.

**Pull request closed without merging**